keps/sig-storage/625-csi-migration/README.md

```diff
@@ -408,6 +408,63 @@ When the feature is not enabled, only provision/deletion/resize should fail.
+  - If there are issues regarding volume attachment. Check the api server VolumeAttachment objects to see if there are any events or error messages there. Check the pod that needs to attach the volume and see if there are error messages/events there.
+    - If the error occurs from CSI driver side, check the CSI driver log and csi-attacher logs to locate the issue.
+    - If the error occurs from Kube-Controller-Manager, check the logs from attach_detach_controller to find useful information.
+    - If everything works fine for attaching, you should see logs like this on Kube-Controller-Manager:
+    ```
+    util.go:313] "CSI Migration: Translate InTree Volume Spec to CSI" VolumeSpec=&{Volume:nil PersistentVolume:&PersistentVolume{ObjectMeta:{xxx
+    operation_generator.go:413] AttachVolume.Attach succeeded for volume "pvc-20b16944-176d-46dc-b164-dd6f9bd07351" (UniqueName: "kubernetes.io/csi/pd.csi.storage.gke.io^projects/UNSPECIFIED/zones/us-central1-b/disks/pvc-20b16944-176d-46dc-b164-dd6f9bd07351") from node "kubernetes-minion-group-c76l"
+    ```
+    - And you should expect logs like these in the csi-attacher, this proves that the VolumeAttachment object is created:
+    ```
+    I0121 17:33:54.751633       1 controller.go:208] Started VA processing "csi-5c520a9e6aa4752597d0fcdc01a4fd5106ae643f3d969d3f4fe875d538f06aa1"
+    I0121 17:33:54.751669       1 csi_handler.go:218] CSIHandler: processing VA "csi-5c520a9e6aa4752597d0fcdc01a4fd5106ae643f3d969d3f4fe875d538f06aa1"
+    I0121 17:33:54.751684       1 csi_handler.go:245] Attaching "csi-5c520a9e6aa4752597d0fcdc01a4fd5106ae643f3d969d3f4fe875d538f06aa1"
+    I0121 17:33:54.751693       1 csi_handler.go:424] Starting attach operation for "csi-5c520a9e6aa4752597d0fcdc01a4fd5106ae643f3d969d3f4fe875d538f06aa1"
+    ```
+  - If there are issues regarding volume mount, check the error message or event from the pod that need to mount the volume.
+    - If the error occurs from the CSI driver side, check the CSI driver log to find the error that has been thrown.
+    - If the error occurs from the Kubelet side, check kubelet logs from the node that the pod was assigned to. Also, there will be logs from Kube-Controller-Manager which related to OperationExecutor and OperationGenerator that might provide some other insights related to the issue.
+    - You should expect logs from Kubelet regarding mount if it succeeded successfully:
+    ```
+    "MountVolume.MountDevice succeeded for volume \"pvc-20b16944-176d-46dc-b164-dd6f9bd07351\" device mount path \"/var/lib/kubelet/plugins/kubernetes.io/csi/pd.csi.storage.gke.io/******\"
+    "MountVolume.SetUp succeeded for volume \"pvc-20b16944-176d-46dc-b164-dd6f9bd07351\"
+    ```
+  - One of the most commonly seen failure scenario would be CSI driver is not installed but CSI migration is enabled. This could lead to volume operation failure. You can find logs like following in Kube-Controller-Manager:
```

Comment:
Note that these logs also show up in successful cases too. In the error case though, the difference is there is no csi driver to act on it so there is no progress.

Reply:
Added a note.